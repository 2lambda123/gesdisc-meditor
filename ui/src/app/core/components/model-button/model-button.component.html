<button color="primary" (click)="goToSearch.emit(model)" fxFlex="100px"
	mat-raised-button style="line-height: inherit; padding: 10px;" >
  <i class="icon-badge icon-badge-lg fa {{model.icon.name}} align-middle" style="margin-bottom: 5px;" [style.background-color]="model.icon.color"></i>
  <div style="font-weight: bold;">{{model.name}}</div>
  <div>({{model.count}})</div>
</button>
