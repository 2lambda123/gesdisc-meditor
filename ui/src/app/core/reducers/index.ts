import { createSelector, createFeatureSelector, ActionReducerMap } from '@ngrx/store';
import * as fromContentTypes from './content-types';

export interface ContentTypesState {
  contentTypes: fromContentTypes.State;
}

export interface State {
  contentTypes: ContentTypesState;
}

export const reducers = {
  contentTypes: fromContentTypes.reducer
};

export const selectContentTypesState = createFeatureSelector<ContentTypesState>('contentTypes');


export const getContentTypesEntitiesState = createSelector(
  selectContentTypesState,
  state => state.contentTypes
);

export const {
  selectIds: getContentTypeIds,
  selectEntities: getContentTypeEntities,
  selectAll: getAllContentTypes,
  selectTotal: getTotalContentTypes,
} = fromContentTypes.adapter.getSelectors(getContentTypesEntitiesState);
// export const selectContentTypesIds = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesIds);
// export const selectContentTypesEntities = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesEntities);
// export const selectAllContentTypes = createSelector(selectContentTypesState, fromContentTypes.selectAllContentTypes);
// export const selectContentTypesTotal = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesTotal);
// export const selectCurrentUserId = createSelector(selectUserState, fromContentTypes.getSelectedUserId);

// export const selectCurrentUser = createSelector(
//   selectUserEntities,
//   selectCurrentUserId,
//   (userEntities, userId) => userEntities[userId]
// );
