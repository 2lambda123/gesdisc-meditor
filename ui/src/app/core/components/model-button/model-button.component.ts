import { Component, Input, Output, EventEmitter, OnInit } from '@angular/core';
import { ModelCatalogEntry } from '../../../service/model/modelCatalogEntry';

@Component({
	selector: 'med-model-button',
	template: `
		<button color="primary" (click)="goToSearch.emit(model)" fxFlex="100px"
			mat-raised-button style="line-height: inherit; padding: 10px;" >
		  <i class="icon-badge icon-badge-lg fa {{model.icon?.name}} align-middle" style="margin-bottom: 5px;" [style.background-color]="model.icon?.color"></i>
		  <div style="font-weight: bold;">{{model.name}}</div>
		  <div>({{model['x-meditor'] && model['x-meditor'].count ? model['x-meditor'].count : 0}})</div>
		</button>
	`
})

export class ModelButtonComponent implements OnInit {

	@Input() public model: ModelCatalogEntry;
	@Output() goToSearch = new EventEmitter<ModelCatalogEntry>();

	constructor () {}

	ngOnInit() {

	}
}
