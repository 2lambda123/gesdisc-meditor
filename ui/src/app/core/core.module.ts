import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { FlexLayoutModule } from '@angular/flex-layout';
// CONTAINERS
import { MainComponent } from './containers/main';
import { NotFoundPageComponent } from './containers/not-found-page';
import { SplashPageContainer } from './containers/splash-page/splash-page.container';
// COMPONENTS
import { ContentTypeBoxComponent } from './components/content-type-button/content-type-button.component';
import { ContentTypeSplashComponent } from './components/splash-box/splash-box.component';
import { LayoutComponent } from './components/layout';
import { NavItemComponent } from './components/nav-item';
import { SidenavComponent } from './components/sidenav';
import { ToolbarComponent } from './components/toolbar';
import { MaterialModule } from '../material';

import { AngularFontAwesomeModule } from 'angular-font-awesome';

import { AuthModule } from '../auth/auth.module';

export const COMPONENTS = [
  MainComponent,
  NotFoundPageComponent,
  SplashPageContainer,
  ContentTypeBoxComponent,
  ContentTypeSplashComponent,
  LayoutComponent,
  NavItemComponent,
  SidenavComponent,
  ToolbarComponent,
];

@NgModule({
  imports: [CommonModule, RouterModule, MaterialModule, FlexLayoutModule, AngularFontAwesomeModule, AuthModule.forRoot()],
  declarations: COMPONENTS,
  exports: COMPONENTS
})

export class CoreModule {
  static forRoot() {
    return {
      ngModule: CoreModule
    };
  }
}
