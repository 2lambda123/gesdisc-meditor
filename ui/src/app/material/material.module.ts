import { NgModule } from '@angular/core';

import {
  MatInputModule,
  MatCardModule,
  MatButtonModule,
  MatSidenavModule,
  MatListModule,
  MatIconModule,
  MatToolbarModule,
  MatProgressSpinnerModule,
  MatDatepickerModule,
  MatCheckboxModule,
  MatSelectModule
} from '@angular/material';

@NgModule({
  imports: [
    MatInputModule,
    MatCardModule,
    MatButtonModule,
    MatSidenavModule,
    MatListModule,
    MatIconModule,
    MatToolbarModule,
    MatProgressSpinnerModule,
    MatDatepickerModule,
    MatCheckboxModule,
  	MatSelectModule
  ],
  exports: [
    MatInputModule,
    MatCardModule,
    MatButtonModule,
    MatSidenavModule,
    MatListModule,
    MatIconModule,
    MatToolbarModule,
    MatProgressSpinnerModule,
    MatDatepickerModule,
    MatCheckboxModule,
  	MatSelectModule
  ],
})
export class MaterialModule {}
