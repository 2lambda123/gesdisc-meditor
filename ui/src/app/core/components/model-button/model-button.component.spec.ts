import { ComponentFixture, TestBed  } from '@angular/core/testing';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { MatCardModule, MatIconModule, MatButtonModule } from '@angular/material';
import { ModelButtonComponent } from './model-button.component';

describe('ModelButtonComponent', () => {
	let fixture: ComponentFixture<ModelButtonComponent>;
	let instance: ModelButtonComponent;

	beforeEach(() => {
		TestBed.configureTestingModule({
			imports: [
				NoopAnimationsModule,
				MatCardModule,
				MatButtonModule,
				MatIconModule
			],
			declarations: [
				ModelButtonComponent
			]
		});

		fixture = TestBed.createComponent(ModelButtonComponent);
		instance = fixture.componentInstance;

	});

	describe('with no data', () => {

		

		it('should throw an error when when compiled', () => {
			expect(function() { fixture.detectChanges() }).toThrow(new TypeError("Cannot read property 'icon' of undefined"));
		});

		it('should not have any initial input', () => {
			expect(instance.model).toBeUndefined();
		});
	});

	describe('with mock data', () => {

		it('should compile with minimum input (model has "name" and "description")', () => {
			instance.model = {
				name: 'test',
				description: 'test'
			}
			fixture.detectChanges();
			expect(fixture).toMatchSnapshot();
		});
	
		it('should compile minimum input + icon', () => {
			instance.model = {
				name: 'test',
				description: 'test',
				icon: {
					name: 'fa-warning',
					color: '#AF4873'
				}
			};
			fixture.detectChanges();
			expect(fixture).toMatchSnapshot();
		});
	});
});
