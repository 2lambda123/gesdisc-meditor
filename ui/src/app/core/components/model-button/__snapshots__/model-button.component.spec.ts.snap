// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelButtonComponent with mock data should compile minimum input + icon 1`] = `
<med-model-button
  goToSearch={[Function EventEmitter]}
  model={[Function Object]}
>
  <button
    class="mat-raised-button mat-primary"
    color="primary"
    fxflex="100px"
    mat-raised-button=""
    ng-reflect-color="primary"
    style="line-height: inherit; padding: 10px;"
  >
    <span
      class="mat-button-wrapper"
    >
      <i
        class="icon-badge icon-badge-lg fa fa-warning align-middle"
        style="margin-bottom: 5px; background-color: rgb(175, 72, 115);"
      />
      <div
        style="font-weight: bold;"
      >
        test
      </div>
      <div>
        (0)
      </div>
    </span>
    <div
      class="mat-button-ripple mat-ripple"
      matripple=""
      ng-reflect-centered="false"
      ng-reflect-disabled="false"
      ng-reflect-trigger="[object HTMLButtonElement]"
    />
    <div
      class="mat-button-focus-overlay"
    />
  </button>
</med-model-button>
`;

exports[`ModelButtonComponent with mock data should compile with minimum input (model has "name" and "description") 1`] = `
<med-model-button
  goToSearch={[Function EventEmitter]}
  model={[Function Object]}
>
  <button
    class="mat-raised-button mat-primary"
    color="primary"
    fxflex="100px"
    mat-raised-button=""
    ng-reflect-color="primary"
    style="line-height: inherit; padding: 10px;"
  >
    <span
      class="mat-button-wrapper"
    >
      <i
        class="icon-badge icon-badge-lg fa  align-middle"
        style="margin-bottom: 5px;"
      />
      <div
        style="font-weight: bold;"
      >
        test
      </div>
      <div>
        (0)
      </div>
    </span>
    <div
      class="mat-button-ripple mat-ripple"
      matripple=""
      ng-reflect-centered="false"
      ng-reflect-disabled="false"
      ng-reflect-trigger="[object HTMLButtonElement]"
    />
    <div
      class="mat-button-focus-overlay"
    />
  </button>
</med-model-button>
`;
